sw/device/tests/sim_dv/ast_clk_rst_inputs.c

```diff
@@ -0,0 +1,577 @@
+}
+
+static void en_plic_irqs(dif_rv_plic_t *plic) {
+  top_earlgrey_plic_irq_id_t plic_irqs[] = {
+      kTopEarlgreyPlicIrqIdAdcCtrlAonMatchDone};
+
+  for (uint32_t i = 0; i < ARRAYSIZE(plic_irqs); ++i) {
+    CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(
+        plic, plic_irqs[i], kTopEarlgreyPlicTargetIbex0, kDifToggleEnabled));
+
+    // Assign a default priority
+    CHECK_DIF_OK(dif_rv_plic_irq_set_priority(plic, plic_irqs[i], 0x1));
+  }
+
+  // Enable the external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+}
+
+void adc_setup(bool first_adc_setup) {
+  // Enable adc interrupts.
+  CHECK_DIF_OK(dif_adc_ctrl_irq_set_enabled(&adc_ctrl, kDifAdcCtrlIrqMatchDone,
+                                            kDifToggleEnabled));
+
+  uint16_t channel0_filter0_max =
+      (kChannel0MaxHighByte << 8) | kChannel0MaxLowByte;
+  uint16_t channel0_filter0_min =
+      (kChannel0MinHighByte << 8) | kChannel0MinLowByte;
+  uint16_t channel1_filter0_max =
+      (kChannel1MaxHighByte << 8) | kChannel1MaxLowByte;
+  uint16_t channel1_filter0_min =
+      (kChannel1MinHighByte << 8) | kChannel1MinLowByte;
+
+  if (first_adc_setup) {
+    // Setup ADC configuration.
+    configure_adc_ctrl(&adc_ctrl);
+  } else {
+    CHECK_DIF_OK(dif_adc_ctrl_reset(&adc_ctrl));
+  }
+
+  en_plic_irqs(&plic);
+  // Setup ADC filters. There is one filter for each channel.
+  CHECK_DIF_OK(dif_adc_ctrl_configure_filter(
+      &adc_ctrl, kDifAdcCtrlChannel0,
+      (dif_adc_ctrl_filter_config_t){.filter = kDifAdcCtrlFilter0,
+                                     .generate_irq_on_match = true,
+                                     .generate_wakeup_on_match = true,
+                                     .in_range = true,
+                                     .max_voltage = channel0_filter0_max,
+                                     .min_voltage = channel0_filter0_min},
+      kDifToggleDisabled));
+  CHECK_DIF_OK(dif_adc_ctrl_configure_filter(
+      &adc_ctrl, kDifAdcCtrlChannel1,
+      (dif_adc_ctrl_filter_config_t){.filter = kDifAdcCtrlFilter0,
+                                     .generate_irq_on_match = true,
+                                     .generate_wakeup_on_match = true,
+                                     .in_range = true,
+                                     .max_voltage = channel1_filter0_max,
+                                     .min_voltage = channel1_filter0_min},
+      kDifToggleDisabled));
+
+  // enable filters.
+  CHECK_DIF_OK(dif_adc_ctrl_filter_set_enabled(
+      &adc_ctrl, kDifAdcCtrlChannel0, kDifAdcCtrlFilter0, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_adc_ctrl_filter_set_enabled(
+      &adc_ctrl, kDifAdcCtrlChannel1, kDifAdcCtrlFilter0, kDifToggleEnabled));
+
+  if (first_adc_setup) {
+    CHECK_DIF_OK(dif_adc_ctrl_set_enabled(&adc_ctrl, kDifToggleEnabled));
+  }
+}
+
+void ast_enter_sleep_states_and_check_functionality(
+    dif_pwrmgr_domain_config_t pwrmgr_config,
```

Comment:
do we also test the jittery clock enable to make sure things are behaving correctly?

Reply:
yes. the jitter is enabled in this test 
https://github.com/lowRISC/opentitan/blob/26c8f284a1cc4895e6a3a3ef98b665c2edcf0174/sw/device/tests/sim_dv/ast_clk_rst_inputs.c#L544
do you prefer to enable the gitter using the flag en_jitter in chip_sim_cfg.hjson file?

Reply:
o looks good, i didn't see that somehow. 
Will you guys have some checks on ast side to make sure jitter is behaving properly? Because we don't have a lot we can do on the open source side.

Reply:
Yes will check on ast side.